Octokit/Exceptions/AbuseException.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Runtime.Serialization;
+using System.Security;
+
+namespace Octokit
+{
+    /// <summary>
+    /// Represents a subset of the HTTP 403 - Forbidden response returned from the API when the forbidden response is related to an abuse detection mechanism.
+    /// Containts the amount of seconds after which it's safe to retry the request.
+    /// </summary>
+#if !NETFX_CORE
+    [Serializable]
+#endif
+    [SuppressMessage("Microsoft.Design", "CA1032:ImplementStandardExceptionConstructors",
+        Justification = "These exceptions are specific to the GitHub API and not general purpose exceptions")]
+    public class AbuseException : ForbiddenException
+    {
+        private readonly int? RetrySecondsDefault = null;
+        /// <summary>
+        /// Constructs an instance of AbuseException
+        /// </summary>
+        /// <param name="response">The HTTP payload from the server</param>
+        public AbuseException(IResponse response) : this(response, null)
+        {
+        }
+
+        /// <summary>
+        /// Constructs an instance of AbuseException
+        /// </summary>
+        /// <param name="response">The HTTP payload from the server</param>
+        /// <param name="innerException">The inner exception</param>
+        public AbuseException(IResponse response, Exception innerException)
+            : base(response, innerException)
+        {
+            Debug.Assert(response != null && response.StatusCode == HttpStatusCode.Forbidden,
+                "AbuseException created with wrong status code");
+
+            SetRetryAfterSeconds(response);
+        }
+
+        private void SetRetryAfterSeconds(IResponse response)
```

Comment:
To me it seems a bit overkill to have a property, default property and 2 private functions, all related to pulling the value out of the headers and parsing it.  I think maintainability would be higher if it was a single function called from the `ctor` that took the `HttpResponse` as input and returned the `int?` parsed value.  It also eliminates the need to declare a `default` value, as the function could just return `null` at the end if all the `TrypGetHeader` `TryParse` and `>= 0` tests didnt satisfy

Im happy to discuss if you feel things are more granularly testable with the current implementation but this was my initial thought...